internal/tiltfile/tiltfile_test.go

```diff
@@ -3854,6 +3854,39 @@ local_resource("toplvl-local", "echo hello world", ["foo/baz", "foo/a.txt"])
 	}, ltTop.LocalRepos())
 }
 
+func TestLocalResourceIgnoreOnly(t *testing.T) {
+	f := newFixture(t)
+	defer f.TearDown()
+
+	f.file("Tiltfile", `
+local_resource("test", "echo hi", ["foo"], ignore="**/*.a", only="**/bar.*")
```

Comment:
i don't understand what the different is between `deps` and `only`? they seem exactly the same to me

Reply:
maybe the new test array makes things clearer?

not sure if this is what you're asking, but the idea is:
`foo.a` matches ignore but not only
`bar.a` matches ignore *and* only
`bar.b` matches only but not ignore

Reply:
also for future reference: ** globs in `only` parameters won't work. they're not implemented correctly in the underlying dockerignore engine. 

This is documented here: https://docs.tilt.dev/api.html#api.docker_build

and our docker_build command will error if you try to use them this way
https://github.com/windmilleng/tilt/blob/master/internal/tiltfile/tiltfile_test.go#L3370

Reply:
> also for future reference: ** globs in only parameters won't work. they're not implemented correctly in the underlying dockerignore engine.

Actually, thinking about this last night. The unit test is passing as-is, which, uh, seems like good evidence to the contrary?

This made me try harder to remember the details of this bug, and IIRC underlying dockerignore engine is perfectly fine, and there's an optimization in the *tarring code* that inappropriately uses dockerignore to skip directories. Since we're not tarring for local resource, it should be fine.

Reply:
talked about this offline, but adding this comment for posterity

the watchmanager won't watch things correctly, and this test doesn't exercise the watch code at all.

all build systems have this problem, it's not unique to us. 

here's the discussion in Docker:
https://github.com/moby/moby/issues/30018

here's the discussion in Bazel:
https://docs.bazel.build/versions/master/be/functions.html#glob

The general problem is that once you start to have complicated sequences of overlapping include globs and ignore globs, they become impossible to implement without stat-ing the whole file system (and causing performance problems that are hard to diagnose), so all build systems put pretty tight restrictions on how you can use them.